components/clp-package-utils/clp_package_utils/general.py

```diff
@@ -556,3 +562,31 @@ def validate_path_for_container_mount(path: pathlib.Path) -> None:
                 f"Invalid path: `{path}` cannot be under '{prefix}' which may overlap with a path"
                 f" in the container."
             )
+
+
+def validate_dataset(clp_config: CLPConfig, input_dataset: Optional[str]) -> Optional[str]:
```

Comment:
Since the function name is "validate"_dataset, I don't think it should return a string. some suggestions have been made in PR1050